ibm_cloud_sdk_core/authenticators/authenticator.py

```diff
@@ -19,15 +19,15 @@
 class Authenticator(ABC):
     """This interface defines the common methods and constants associated with an Authenticator implementation."""
     @abstractmethod
-    def authenticate(self, req: dict):
+    def authenticate(self, req: dict) -> None:
         """Perform the necessary authentication steps for the specified request.
 
```

Comment:
I think you should document here that the `req` parameter will be modified to contain the appropriate authentication information.